src/Processors/QueryPlan/PartsSplitter.h

```diff
@@ -0,0 +1,322 @@
+#pragma once
+
+#include <algorithm>
+#include <numeric>
+#include <queue>
+#include <set>
+#include <vector>
+
+#include <Core/Field.h>
+#include <Parsers/ASTFunction.h>
+#include <Parsers/ASTIdentifier.h>
+#include <Storages/MergeTree/RangesInDataPart.h>
+
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int LOGICAL_ERROR;
+}
+
+struct IIndexAccess
```

Comment:
Generally, it's good to write a comment to
* every class (both interface and implementation)
* all virtual methods
* and, actually, every method which purpose is not obvious from the first glance

Reply:
Looks like this iface has single implementation.
Do we actually need it? I think that, generally, you should not create an interfece if you don't plan to have > 1 implementations...